packages/web-components/fast-ssr/src/template-renderer/template-renderer.ts

```diff
@@ -31,6 +40,110 @@ export class TemplateRenderer implements Readonly<TemplateRendererConfiguration>
         renderInfo: RenderInfo,
         source?: unknown
     ): IterableIterator<string> {
-        yield "";
+        const codes = parseTemplateToOpCodes(template);
+
+        for (const code of codes) {
+            switch (code.type) {
+                case OpType.text:
+                    yield code.value;
+                    break;
+                case OpType.directive: {
+                    const { directive } = code;
+                    const ctor = directive.constructor;
```

Comment:
@EisenbergEffect @janechu @williamw2 I wanted to get some early PR feedback on the directive plugin design (This is the ability to have SSR specific rendering for directive implementations). The basic idea is that a [DirectiveRenderer](packages/web-components/fast-ssr/src/template-renderer/directives.ts) is created, which holds a reference to the constructable directive to use as a registry key. Then, when we encounter a directive instance while rendering out a template, we get the *constructor* of the encountered directive instance and check to see if we have a renderer registered for that constructor. I've designed it this way to keep lookup times fast, but there are a few drawbacks I want to get thoughts on:
1. A directive *must* be a ES6 class
2. constructor comparison is rigid and won't match something that *inherits* from the directive implementation, which means an author can't have a single renderer that handles multiple directives without specifically registering them.

A few other approaches I considered were using a test function that returns a bool when the directive should be handled by the renderer and using `instanceof ` checks instead of constructor equality, but the downside to those approaches is lookup times will be slower. 

Do you all think the above limitations are acceptable? Is there another way to approach this I'm not considering that would be just as fast but also more flexible?

Reply:
I think this is fine.